crates/orchestrator/src/spawner.rs

```diff
@@ -150,7 +151,11 @@ where
     node.rpc_port.drop_listener();
     node.prometheus_port.drop_listener();
 
-    let running_node = ctx.ns.spawn_node(spawn_ops).await?;
+    let err_context = format!(
+        "Failed to spawn node: {} with opts: {:#?}",
+        node.name, spawn_ops
+    );
+    let running_node = ctx.ns.spawn_node(spawn_ops).await.context(err_context)?;
 
```

Comment:
might be worth refactoring this spawn_node so it takes a ref option instead so that you can lazily compute the context here.

Reply:
Make sense, let me change it. Thx!